test/content/support.js

```diff
@@ -202,39 +202,25 @@ function resetDB() {
  * Equivalent to JSON.stringify, except that object properties are stringified
  * in a sorted order.
  */
-function stableStringify(obj, level, label) {
-	if (!level) level = 0;
-	let indent = '\t'.repeat(level);
-	
-	if (label) label = JSON.stringify('' + label) + ': ';
-	else label = '';
-	
-	if (typeof obj == 'function' || obj === undefined) return null;
-	
-	if (typeof obj != 'object' || obj === null) return indent + label + JSON.stringify(obj);
-	
-	if (Array.isArray(obj)) {
-		let str = indent + label + '[';
-		for (let i=0; i<obj.length; i++) {
-			let json = stableStringify(obj[i], level + 1);
-			if (json === null) json = indent + '\tnull'; // function
-			str += '\n' + json + (i < obj.length-1 ? ',' : '');
+function stableStringify(obj) {
+	function recompose (obj) {
+		var keys = Object.keys(obj);
+		keys.sort();
+		var o = {};
+		for (let key in obj) {
+			o[key] = obj[key];
+			delete obj[key];
+		}
+		for (let i=0,ilen=keys.length;i<ilen;i++) {
+			obj[keys[i]] = o[keys[i]];
```

Comment:
While this generally works, it's not guaranteed and something like the following yields incorrect sorting (though still technically deterministic/stable):
```javascript
var k = ["1", "1a", "2", "2a", "bar", "foo"].sort();
// [ "1", "1a", "2", "2a", "bar", "foo" ]

var o = {};
for (var i=0; i<k.length; i++) {
  o[k[i]] = true;
}

JSON.stringify(o);
// "{"1":true,"2":true,"1a":true,"2a":true,"bar":true,"foo":true}"
```

Since it's still stable and doesn't break anything currently, I don't particularly have much against it, except that, because the order of `JSON.stringify` is still not guaranteed, whether this remains stable or not is out of our control.